### PR TITLE
Adjust parsing of base_url when constructing Pulp PyPI

### LINKED ISSUE #33: Newly registered Python indexes are wrong
**Describe the bug**

```
 {
  "url": "https://pulp.operate-first.cloud/https://pulp-api-5c5f4669d8-qgvvh/pypi/gym-donkeycar/simple"
 },
 {
  "url": "https://pulp.operate-first.cloud/https://pulp-api-5c5f4669d8-qgvvh/pypi/test/simple"
 },
 {
  "url": "https://pulp.operate-first.cloud/https://pulp-api-5c5f4669d8-qgvvh/pypi/hello-world/simple"
```

**To Reproduce**
1. Go to stage environment
2. Check registered Python indexes available on Pulp
3. See the error

**Expected behavior**

Index URLs registered should be valid.

### PR DESCRIPTION
**Pull request by fridex**
## Related Issues and Dependencies

Fixes: #33

## This introduces a breaking change

- [x] No

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -46,10 +47,10 @@ def _list_pulp_python_repositories(instance_host: str, *, username: str, passwor
     response.raise_for_status()
 
     for result in response.json()["results"]:
-        simple_repository = (
-            f"{result['base_url']}simple" if result["base_url"].endswith("/") else f"{result['base_url']}/simple"
+        path = urlparse(result["base_url"]).path
+        yield urlunparse(
+            ("https", instance_host, f"{path}/simple" if not path.endswith("/") else f"{path}simple", "", "", "")
```
Comment (Gregory-Pereira) [inline app.py:52]: `instance_host` should be also slash validated too shouldn't it?
```suggestion
            ("https", f"{instance_host}" if not instance_host.endswith("/") else f"{instance_host[0:-1]}", f"{path}/simple" if not path.endswith("/") else f"{path}simple", "", "", "")
```

**Comment by Gregory-Pereira** [comment]
I just assumed that when this was called the `instance_host` would be a full URL (ex: `https://pulp.operate-first.cloud/`). If we dont want to add more validation for the `instance_host` and make sure `https://` is not already included in `instance_host` the help command or base `README.md` could be adjusted to discuss this.

**Comment by fridex** [comment]
> I just assumed that when this was called the `instance_host` would be a full URL (ex: `https://pulp.operate-first.cloud/`). If we dont want to add more validation for the `instance_host` and make sure `https://` is not already included in `instance_host` the help command or base `README.md` could be adjusted to discuss this.

That should be the difference between [host](https://en.wikipedia.org/wiki/Host_(network)) and [URL](https://en.wikipedia.org/wiki/URL)[](url).

**Comment by Gregory-Pereira** [comment]
> > I just assumed that when this was called the `instance_host` would be a full URL (ex: `https://pulp.operate-first.cloud/`). If we dont want to add more validation for the `instance_host` and make sure `https://` is not already included in `instance_host` the help command or base `README.md` could be adjusted to discuss this.
> 
> That should be the difference between [host](https://en.wikipedia.org/wiki/Host_(network)) and [URL](https://en.wikipedia.org/wiki/URL).

Got it. In that case /approve 👍

**Review by Gregory-Pereira — APPROVED** [review]
👍  /approve

**Comment by fridex** [comment]
> +1 /approve

Thanks!

/approve

**Pull request merged**